src/NServiceBus.Core/HeaderDictionary.cs

```diff
@@ -0,0 +1,78 @@
+// unset
+
+#pragma warning disable 1591
+namespace NServiceBus
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    ///
+    /// </summary>
+    public class HeaderDictionary
```

Comment:
should there also be a ctor that accepts a readonly dictionary which would allow us to change the `MessageContext` headers type?